### PR TITLE
chore: fix some unlinted errors

### PR DESCRIPTION
**Pull request by saintube**
### Ⅰ. Describe what this PR does

chore: fix the found errors for the license header and deadcode. also, revise the logs in the podthrottled collector and metric cache.

These errors occasionally are linted when using the linter with a higher version. However, the linter's upgrade depends on the Golang version. e.g. The golangci-lint v1.51.2 requires go1.19. So let's separate the issues and just fix the lint errors in the current PR.

### V. Checklist

- [ ] I have written necessary docs and comments
- [ ] I have added necessary unit tests and integration tests
- [ ] All checks passed in `make test`

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -58,7 +58,7 @@ linters:
   fast: false
   disable-all: true
   enable:
-    - deadcode
+    - deadcode # unused
```
Comment (jasonliu747) [inline .golangci.yml:61]: 这个可以去掉了，换成unused

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -20,7 +20,7 @@ jobs:
       - run: sudo apt update && sudo apt install -y libpfm4 libpfm4-dev
       - uses: golangci/golangci-lint-action@v3
         with:
-          version: v1.47.3
+          version: v1.51.2
```
Comment (jasonliu747) [inline .github/workflows/ci.yaml:23]: 如果没有适配问题，可以直接换成1.55.2

File: Makefile [hunk 3]
```diff
@@ -218,7 +218,7 @@ HACK_DIR ?= $(PWD)/hack
 ## Tool Versions
 KUSTOMIZE_VERSION ?= v3.8.7
 CONTROLLER_TOOLS_VERSION ?= v0.9.0
-GOLANGCILINT_VERSION ?= v1.47.3
+GOLANGCILINT_VERSION ?= v1.51.2
```
Comment (jasonliu747) [inline Makefile:221]: v1.55.2

File: .golangci.yml (hunk 1 above)
Comment (saintube) [reply .golangci.yml]: 两者的默认行为有一些区别，unused 校验范围更大，单独提PR处理吧

**Comment by eahydra** [comment]
内部的CI执行时也都OK的么？

**Comment by saintube** [comment]
> 内部的CI执行时也都OK的么？

CI 的 linter 版本目前都是 Makefile 里指定的

**Review by eahydra — APPROVED** [review]
/lgtm
/approve

**Pull request merged**